src/Form/matching.js

```diff
@@ -500,24 +504,150 @@ class Matching {
     }
 }
 
+/**
+ *  @returns {SupportedTypes}
+ */
+function getInputType (input) {
+    const attr = input.getAttribute(ATTR_INPUT_TYPE)
+    if (isValidSupportedType(attr)) {
+        return attr
+    }
+    return 'unknown'
+}
+
+/**
+ * Retrieves the main type
+ * @param {SupportedTypes} type
+ * @returns {SupportedMainTypes}
+ */
+function getMainTypeFromType (type) {
+    const mainType = type.split('.')[0]
+    switch (mainType) {
+    case 'credentials':
+    case 'creditCard':
+    case 'identities':
+        return mainType
+    }
+    return 'unknown'
+}
+
 /**
  * Retrieves the input main type
  * @param {HTMLInputElement} input
- * @returns {SupportedMainTypes | string}
+ * @returns {SupportedMainTypes}
  */
 const getInputMainType = (input) =>
-    input.getAttribute(ATTR_INPUT_TYPE)?.split('.')[0] ||
-    'unknown'
+    getMainTypeFromType(getInputType(input))
+
+/** @typedef {supportedIdentitiesSubtypes[number]} SupportedIdentitiesSubTypes */
+const supportedIdentitiesSubtypes = /** @type {const} */ ([
+    'emailAddress',
+    'cardName',
```

Comment:
`cardName` shouldn't go here, but in the credit card array (it's already there, so just remove here).